cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/remote/RemotePrivilegesFetcherTest.java

```diff
@@ -77,18 +77,28 @@ public static void setup() throws IOException {
     Location externalAuthJar = AppJarHelper.createDeploymentJar(locationFactory, InMemoryAuthorizer.class, manifest);
     cConf.set(Constants.Security.Authorization.EXTENSION_JAR_PATH, externalAuthJar.toString());
     Injector injector = Guice.createInjector(new AppFabricTestModule(cConf));
-    txManager = injector.getInstance(TransactionManager.class);
-    txManager.startAndWait();
-    datasetService = injector.getInstance(DatasetService.class);
-    datasetService.startAndWait();
     remoteSysOpService = injector.getInstance(RemoteSystemOperationsService.class);
     remoteSysOpService.startAndWait();
-    appFabricServer = injector.getInstance(AppFabricServer.class);
-    appFabricServer.startAndWait();
+    waitForService(injector.getInstance(DiscoveryServiceClient.class));
     authorizerInstantiator = injector.getInstance(AuthorizerInstantiator.class);
     privilegesFetcher = injector.getInstance(PrivilegesFetcher.class);
   }
 
+  private static void waitForService(DiscoveryServiceClient discoveryService) throws InterruptedException {
+    final CountDownLatch startLatch = new CountDownLatch(1);
+    discoveryService.discover(Constants.Service.REMOTE_SYSTEM_OPERATION)
+      .watchChanges(new ServiceDiscovered.ChangeListener() {
```

Comment:
@bdmogal Can you make the above change to the RemoteNamespaceQueryHandlerTest as well?